priv/samples/basics/spi_rgb_leds.livemd

```diff
@@ -0,0 +1,371 @@
+
+| Gray Scale | Brigtness |   Blue    |   Green   |    Red    |
+|:----------:|:---------:|:---------:|:---------:|:---------:|
+|   111      |   11111   | 1111 1111 | 1111 1111 | 1111 1111 |
+|   3 Bits   |   5 Bits  |   8 Bits  |   8 Bits  |   8 Bits  |
+
+Elixir example: `<<3::3, 1::5, 0xFF, 0xFF, 0xFF>>`.
+
+###### End Frame
+
+|   0   |
+| :---: |
+|  0x0  |
+| 1 Byte |
+
+Times (n/2), where n is the number of LEDs on the strip rounded to the next byte.
+
+Elixir example: `<<0::8>>`.
+
+[return to top](#table-of-contents)
+
+## Blinking a single LED
+
+First, we will turn on your first LED.
+
+Start frame is static, and can be reused.
+
+We will only have one LED frame, to blink the first LED on the strip.
+
+However, our end frame depends on the number of LEDs we have, 
+so lets calculate the number of bytes we will end with.
+
+The `Kernel.<>/2` operator concatinates our binaries.
+
+```elixir
+led_count = 1
+start_frame = << 0::32 >>
+# The end frame is 8 '0' bits * (leds - 1) / 16 (rounded up)
+# These are clock edges to ensure the LED's respond immediately and is borrowed from the FASTLed library.
+end_frame = 
+  Enum.map(1..round((led_count - 1) / 16), fn _ -> <<0::8>> end)
+  |> Enum.map_join(& &1)
+
+# Creating a List of frames with each LED on white
+led_frame = <<7::3, 1::5, 0xff, 0xff, 0xff>>
+
+# Transfer the frames over spi
+Circuits.SPI.transfer(ref, start_frame <> led_frame <> end_frame)
+```
+
+Your first LED should bet lit up white now.
+
+Lets blink it a few times, alternating the `led_frame` and an `off_frame`.
+
+```elixir
+blink_times = 10
+delay = 1000
+off_frame = <<7::3, 1::5, 0x00, 0x00, 0x00>>
+
+Enum.map(0..blink_times, fn x ->
+    case rem(x, 2) do
+        0 -> Circuits.SPI.transfer(ref, start_frame <> led_frame <> end_frame)
+        _ -> Circuits.SPI.transfer(ref, start_frame <> off_frame <> end_frame)
+    end
+    Process.sleep(delay)
+end)
+```
+
+[return to top](#table-of-contents)
+
+## Managing all the LEDs on a strip
+
+We can create lists of frames and send them all at one.
+
+Update the `led_count` to reflect your strip.
+
+```elixir
+# Set the number of LED's in your strip
+led_count = 144
+
+# The end frame is 8 '0' bits * (leds - 1) / 16 (rounded up)
+# These are clock edges to ensure the LED's respond immediately and is borrowed from the FASTLed library.
+end_frame =
+  Enum.map(1..round((led_count - 1) / 16), fn _ -> <<0::8>> end)
+  |> Enum.map_join(& &1)
+
+# Creating a List of frames with each LED on white
+led_frames = Enum.map(1..led_count, fn _ -> led_frame end)
+off_frames = Enum.map(1..led_count, fn _ -> off_frame end)
+
+# Transfer the frames over spi
+Circuits.SPI.transfer(ref, start_frame <> Enum.join(led_frames) <> end_frame)
+
+# Wait for a prespecified duration of time.
+Process.sleep(delay)
+
+# Turn them off
+Circuits.SPI.transfer(ref, start_frame <> Enum.join(off_frames) <> end_frame)
+```
+
+[return to top](#table-of-contents)
+
+## Simple larson scanner
```

Comment:
```suggestion
## Simple Larson scanner
```

Is Larsen capitalized since it's a person's name?

Reply:
:100: